Source/EventAnalysis/KTCollinearTrackClustering.hh

```diff
@@ -0,0 +1,127 @@
+/**
+ @file KTCollinearTrackClustering.hh
+ @brief Contains KTCollinearTrackClustering
+ @details Groups collinear tracks into one
+ @author: E. Zayas
+ @date: May 1, 2017
+ */
+
+#ifndef KTCOLLINEARTRACKCLUSTERING_HH_
+#define KTCOLLINEARTRACKCLUSTERING_HH_
+
+#include "KTPrimaryProcessor.hh"
+
+#include "KTSlot.hh"
+#include "KTData.hh"
+#include "KTMemberVariable.hh"
+#include "KTProcessedTrackData.hh"
+
+#include <algorithm>
+#include <set>
+#include <vector>
+#include <list>
+
+
+namespace Katydid
+{
+    
+    /*!
+     @class KTCollinearTrackClustering
+     @author E. Zayas
+
+     @brief Identifies and groups collinear tracks
+
+     @details
+     Clusters tracks together in the slope/intercept plane with an elliptical boundary
+
+     Configuration name: "collinear-track-clustering"
+
+     Available configuration values:
+     - "slope-radius": clustering tolerance in the slope axis
+     - "frequency-radius": clustering tolernace in the intercept axis
+
+     Slots:
+     - "track": void (shared_ptr<KTData>) -- If this is a new acquisition; Adds tracks to the internally-stored set of points; Requires KTProcessedTrackData.
+     - "do-clustering": void () -- Triggers clustering algorithm
+
+     Signals:
+     - "track": void (shared_ptr<KTData>) -- Emitted for each group found; Guarantees KTProcessedTrackData.
+     - "tracks-done": void () -- Emitted when track clustering is complete
+    */
+
+    class KTCollinearTrackClustering : public Nymph::KTPrimaryProcessor
+    {
+        public:
+            KTCollinearTrackClustering(const std::string& name = "collinear-track-clustering");
+            virtual ~KTCollinearTrackClustering();
+
+            bool Configure(const scarab::param_node* node);
+
+            MEMBERVARIABLE(double, SlopeRadius);
+            MEMBERVARIABLE(double, FrequencyRadius);
+
+        public:
+            // Store point information locally
+            bool TakeTrack(KTProcessedTrackData& track);
+
+            void SetNComponents(unsigned nComps);
+            bool DoClustering();
+            bool Run();
+
+
+        private:
+            bool FindCollinearTracks();
+            void FindCluster();
+            void ProcessNewTrack( KTProcessedTrackData& myNewTrack );
+            int GetNUngrouped();
+
+            std::vector< TrackSet > fCompTracks; // input tracks
+
+            std::vector< double > fSlopes;
+            std::vector< double > fIntercepts;
+            std::vector< double > fTimeLengths;
+            std::vector< double > fStartTimes;
+            std::vector< int > fGroupingStatuses;
+
```

Comment:
1. Member variables should not be initialized in the header file.  While legal in C++, it can create unnecessary dependencies, extra recompilation, and other annoyances.  It's better to initialize everything in the constructor's initialization list.

1. Member variable names should start with `f` and be CamelCase after that.  Names to fix: `fUNGROUPED`, `fGROUPED`, `fREMOVED`, `avgQ`, `avgF`, `totalVariance`, `delta`, `bestDelta`, `worstTrack`, `nUngrouped`.

Reply:
Is there any other naming / initialization convention I should follow for constants that are meant to represent a state? For example there are things like `FFTW_FORWARD` and `kTrue` in other libraries - this is the purpose of `fUNGROUPED`, `fGROUPED` and `fREMOVED`. Right now I have them named in all caps after the `f`, and initialized in the header (I have them instead as `const int`s in my local copy now).